source/extensions/filters/http/common/compressor/compressor.cc

```diff
@@ -118,7 +118,6 @@ Http::FilterHeadersStatus CompressorFilter::encodeHeaders(Http::ResponseHeaderMa
     compressor_ = config_->makeCompressor();
   } else if (!skip_compression_) {
     skip_compression_ = true;
-    config_->stats().not_compressed_.inc();
```

Comment:
I don't think this actually gets incremented twice as in the other case skip_compression_ is true?

Reply:
Bah, you are right -- sorry :-( 

Quick drive by diff while I was debugging something else. Sorry folks.